runtime/ops/os/sys_info.rs

```diff
@@ -0,0 +1,281 @@
+    if !NT_SUCCESS(unsafe {
+      RtlGetVersion(&mut version_info as *mut RTL_OSVERSIONINFOEXW as *mut _)
+    }) {
+      String::from("")
+    } else {
+      format!(
+        "{}.{}.{}",
+        version_info.dwMajorVersion,
+        version_info.dwMinorVersion,
+        version_info.dwBuildNumber
+      )
+    }
+  }
+}
+
+pub fn hostname() -> String {
+  #[cfg(target_family = "unix")]
+  // SAFETY: `sysconf` returns a system constant.
+  unsafe {
+    let buf_size = libc::sysconf(libc::_SC_HOST_NAME_MAX) as usize;
+    let mut buf = vec![0u8; buf_size + 1];
+    let len = buf.len();
+    if libc::gethostname(buf.as_mut_ptr() as *mut libc::c_char, len) < 0 {
+      return String::from("");
+    }
+    // ensure NUL termination
+    buf[len - 1] = 0;
+    std::ffi::CStr::from_ptr(buf.as_ptr() as *const libc::c_char)
+      .to_string_lossy()
+      .to_string()
+  }
+  #[cfg(target_family = "windows")]
+  {
+    use std::ffi::OsString;
+    use std::os::windows::ffi::OsStringExt;
+    use winapi::um::winsock2::GetHostNameW;
+
+    let namelen = 256;
+    let mut name: Vec<u16> = std::iter::repeat(0).take(namelen).collect();
+    let err =
+      unsafe { GetHostNameW(name.as_mut_ptr(), namelen as libc::c_int) };
+
+    if err == 0 {
+      // TODO(@littledivy): Probably not the most efficient way.
+      let len = name.iter().take_while(|&&c| c != 0).count();
+      OsString::from_wide(&name[..len])
+        .to_string_lossy()
+        .into_owned()
+    } else {
+      String::from("")
+    }
+  }
+}
+
+#[derive(serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MemInfo {
+  pub total: u64,
+  pub free: u64,
+  pub available: u64,
+  pub buffers: u64,
+  pub cached: u64,
+  pub swap_total: u64,
+  pub swap_free: u64,
+}
+
+pub fn mem_info() -> Option<MemInfo> {
+  let mut mem_info = MemInfo {
+    total: 0,
+    free: 0,
+    available: 0,
+    buffers: 0,
+    cached: 0,
+    swap_total: 0,
+    swap_free: 0,
+  };
+  #[cfg(target_os = "linux")]
+  {
+    fn read_meminfo(mem_info: &mut MemInfo) -> Result<(), AnyError> {
+      let mut s = String::new();
+      File::open("/proc/meminfo")?.read_to_string(&mut s)?;
+      for line in s.lines() {
+        let mut split_line = line.split_whitespace();
+        let label = split_line.next();
+        let value = split_line.next();
+        if value.is_some() && label.is_some() {
+          let label = label.unwrap().split(':').nth(0)?;
+          let value = value.unwrap().parse::<u64>()?;
+          match label {
+            "MemTotal" => mem_info.total = value,
+            "MemFree" => mem_info.free = value,
+            "MemAvailable" => mem_info.available = value,
+            "Buffers" => mem_info.buffers = value,
+            "Cached" => mem_info.cached = value,
+            "SwapTotal" => mem_info.swap_total = value,
+            "SwapFree" => mem_info.swap_free = value,
+            _ => (),
+          }
+        }
```

Comment:
Can break the parser out into a separate function that operates on a string, and add a unit test on it?